src/cript/api/api.py

```diff
@@ -546,30 +547,19 @@ def _internal_save(self, node, known_uuid: Optional[Set[str]] = None) -> Optiona
         # if it does exist we use `patch` if it doesn't `post`.
         node_known = len(self.search(type(node), SearchModes.UUID, str(node.uuid)).current_page_results) == 1
         if node_known:
-            response: Dict = requests.patch(url=f"{self._host}/{node.node_type.lower()}", headers=self._http_headers, data=json_data).json()
+            response: Dict = requests.patch(url=f"{self._host}/{node.node_type.lower()}/{str(node.uuid)}", headers=self._http_headers, data=json_data).json()
         else:
             response: Dict = requests.post(url=f"{self._host}/{node.node_type.lower()}", headers=self._http_headers, data=json_data).json()  # type: ignore
 
-        # If we get a Bad UUID we need to handle this extra and save the Bad node first.
-        if response["code"] == 400 and response["error"].startswith("Bad uuid:"):
-            # We extract the
-            missing_uuid = response["error"][len("Bad uuid: ") : -len(" provided")]
-
-            # Use the find_children functionality to find that node in our current tree
-            # We can have multiple occurrences of the node,
-            # but it doesn't matter which one we save
-            # TODO some error handling, for the BUG case of not finding the UUID
-            missing_node = node.find_children({"uuid": missing_uuid})[0]
-
-            # Now we save the bad node extra.
-            # So it will be known when we attempt to save the graph again.
-            known_uuid.union(self._internal_save(missing_node, known_uuid))  # type: ignore
-            # Since we pre-saved this node, we want it to be UUID edge only the next JSON.
-            # So we add it to the list of known nodes
-            known_uuid.add(missing_uuid)
-            # Recursive call.
-            # Since we should have fixed the "Bad UUID" now, we can try to save the node again
-            return self._internal_save(node, known_uuid)
+        # If we get an error we may be able to fix, we to handle this extra and save the bad node first.
+        # Errors with this code, may be fixable
+        if response["code"] in (400, 409):
```

Comment:
I suggest having a variable for the list of errors that we handle, in case we want to add or subtract to it we can do so later

```suggestion

http_status_code_handled = (400, 409)

# If we get an error we may be able to fix, we to handle this extra and save the bad node first.
# Errors with this code, may be fixable
if response["code"] in http_status_code_handled
```

Reply:
I don't know if we want to rely on the status codes because `HTTP 400` just means generic bad request and can be used for many things. I think this is good for now but at some point we'll need to have solid errors that we handle and we can rely on. Those errors have to for sure be variables for us to be able to change easily if the backend changes

Reply:
okay that is fair

Reply:
I would still suggest to put the error codes in a variable, but I can be okay with it if we decide not to do that for now

```python
http_status_code_handled = (400, 409)
# If we get an error we may be able to fix, we to handle this extra and save the bad node first.
# Errors with this code, may be fixable
if response["code"] in http_status_code_handled
```

Reply:
I see your logic, usually I make it into variables because it makes it easier to update in the future, but I think it is fine for now and we can update it later